js/symbol/get_anchors.js

```diff
@@ -68,7 +74,9 @@ function resample(line, offset, spacing, angleWindowSize, maxAngle, labelLength,
                 x = interpolate(a.x, b.x, t),
                 y = interpolate(a.y, b.y, t);
 
-            if (x >= 0 && x < tileExtent && y >= 0 && y < tileExtent) {
+            if (x >= 0 && x < tileExtent && y >= 0 && y < tileExtent &&
+                    markedDistance - halfLabelLength >= 0 &&
+                    markedDistance + halfLabelLength <= lineLength) {
```

Comment:
This statement could use some more documentation, either in the form of a `// comment` or factored out subexpressions